src/components/Chat.js

```diff
@@ -313,30 +336,42 @@ export default {
 				if (event) {
 					event.preventDefault();
 				}
-				this.$http.post(this.$store.state.apiBaseUrl + '/upload', this.fileUploadFormData).then(response => {
-					console.log('yay', response);
-					let hash = this.proof.hash;
-					this.proof.ipfsHash = response.body.hash;
-					this.addMessage({
-						sender: MessageSenderEnum.APP,
-						primary: true,
-						body: {
-							type: MessageBodyTypeEnum.TEXT,
-							text: 'Note: The file will be visible to any person with access to the hash.'
-						}
-					});
-					this.machine.transition('showSummary');
-				}, response => {
-					console.log('nay', response);
-					this.addMessage({
-						sender: MessageSenderEnum.APP,
-						primary: true,
-						body: {
-							type: MessageBodyTypeEnum.TEXT,
-							text: 'Something went wrong'
-						},
-					});
-				});
+        
+
+        this.getImageDataUrl(this.fileUploadFormData.get('file'), (err, dataUrl) => {
+          console.log('getImageDataUrl', err);
+          if (err) {
+            return callback(err);
+          }
+          this.proof.dataUrl = dataUrl;
+
+          this.addIpfsContent(Buffer.from(dataUrl),(ipfsHash)=>{
```

Comment:
why do you store the dataUrl and not the file itself?

Reply:
since the existing code converted and used base64 images, I thought it prudent to just store them that way.

Reply:
I see. This is just preference. I just thought it's easier to then later get the file by url like (https://ipfs.io/ipfs/<ipfsHash>)

Reply:
Gateways are convenient but introduces centralization. If gateway is down or attacked, we stop working. I think it's a little more robust to pull directly from the DHT. since that's essentially what we're doing with a gateway anyways, just through an intermediary.